File: ums_backend/router/sys_user.go
package router

import (
	"ums_backend/api"

	"github.com/gin-gonic/gin"
)

func InitUserRouter(Router *gin.RouterGroup) {
	userRouter := Router.Group("user")
	{
		userRouter.POST("login", api.LoginUser)
		userRouter.GET("getUserInfo", api.GetUserInfo)
		userRouter.POST("logout", api.LogOut)
		userRouter.POST("getUserList", api.GetSysUserList)
		userRouter.POST("AddSysUser", api.AddSysUser)
		userRouter.POST("getExistUser", api.GetSysExistUser)
		userRouter.POST("ResetSysPassword", api.ResetSysPassword)
	}
}

File: ums_frontend/src/api/sys-user.js
import request from '@/utils/request'

export function login(data) {
  return request({
    url: '/user/login',
    method: 'post',
    data
  })
}

export function getUserInfo(username) {
  return request({
    url: '/user/getUserInfo',
    method: 'get',
    params: { username }
  })
}

export function getUserList(data) {
  return request({
    url: '/user/getUserList',
    method: 'post',
    data
  })
}

export function logout() {
  return request({
    url: '/user/logout',
    method: 'post'
  })
}

export function changeUserStatus() {}

export function createUser(data) {
  return request({
    url: '/user/AddSysUser',
    method: 'post',
    data
  })
}

export function getExistUser(data) {
  return request({
    url: '/user/getExistUser',
    method: 'post',
    data
  })
}

export function resetUserPassword(data) {
  return request({
    url: '/user/ResetSysPassword',
    method: 'post',
    data
  })
}

